backend/src/main/java/ai/verta/modeldb/configuration/RunLiquibaseSeparate.java

```diff
@@ -0,0 +1,36 @@
+package ai.verta.modeldb.configuration;
+
+import ai.verta.modeldb.common.CommonConstants;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class RunLiquibaseSeparate implements Condition {
+  private static final Logger LOGGER = LogManager.getLogger(RunLiquibaseSeparate.class);
+
+  public RunLiquibaseSeparate() {}
+
+  public boolean liquibaseRunSeparately() {
+    var liquibaseRunSeparately =
+        Boolean.parseBoolean(
+            Optional.ofNullable(System.getenv(CommonConstants.RUN_LIQUIBASE_SEPARATE))
+                .orElse("false"));
+    LOGGER.info("RUN_LIQUIBASE_SEPARATE: {}", liquibaseRunSeparately);
+    return false;
+  }
+
+  @Override
+  public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+    return liquibaseRunSeparately();
+  }
+
+  public static class InversedRunLiquibaseSeparate extends RunLiquibaseSeparate {
```

Comment:
it is not about migration. If `liquibaseRunSeparately=true` then it will decide to run only Liquibase and migration separately from the actual service and it will not run the registry backend but if `liquibaseRunSeparately=false` then it will execute registry backend and not run the Liquibase.

Reply:
Ok, All confusion is due to the name of the flags. 
So we have two flags with two different conditions.

1. ENABLE_LIQUIBASE_MIGRATION_ENV_VAR
- This flag will decide whether to run the Liquibase migration and the code migrations or not

2. RUN_LIQUIBASE_SEPARATE
- This flag will decide whether to execute the above process (first point) separately from the actual registry backend or execute it with the actual registry backend.

**Usecases:**  
1. If I want to run the Liquibase migration separately from the actual registry backend then I will set the above two flags like 
```
ENABLE_LIQUIBASE_MIGRATION_ENV_VAR = true
RUN_LIQUIBASE_SEPARATE = true
```
Which is currently we are using in our cluster setup I think. where in the prehook of Kubernetes pod we are setting flags like 
```
ENABLE_LIQUIBASE_MIGRATION_ENV_VAR = true
RUN_LIQUIBASE_SEPARATE = true
```

and then when the actual backend service pod will start after webhook we will set those false like 
```
ENABLE_LIQUIBASE_MIGRATION_ENV_VAR = false
RUN_LIQUIBASE_SEPARATE = false
```

2. Now in the second scenario, let's say I do not want to configure prehook but still I have to run the migration at that time we will not execute migration separately so instead of that we will execute it with our main service. at that time those flags should be like 
```
ENABLE_LIQUIBASE_MIGRATION_ENV_VAR = true
RUN_LIQUIBASE_SEPARATE = false
```
In this scenario, I am using this for my local setup as well because I do not have any prehook steps like Kubernetes.

Reply:
Sure. All of that is correct. I just want to make sure we name our classes/conditions something that makes it clear what they are for, rather than starting with the word "Inverse", which is very confusing and hard to understand the purpose of.

Reply:
Renamed it